manifests/charts/gateway/templates/deployment.yaml

```diff
@@ -64,8 +64,10 @@ spec:
             allowPrivilegeEscalation: false
             privileged: false
             readOnlyRootFilesystem: true
+            {{- if not (and .Values.global (eq .Values.global.platform "openshift")) }}
```

Comment:
This chart doesn't use the `.global` anywhere which is largely the legacy model. I feel like we should probably not introduce this and put it under `.Values.platform` directly instead?

Reply:
so, we'd need this first? https://github.com/istio/istio/pull/50113

Reply:
@howardjohn In this case, the profile is useless, as it sets `global.platform`. If we set `platform` directly in the gateway values, why do we need to copy the helm profiles to every chart?

Reply:
I mean, if the idea is to be able to use "--set profile=openshift" regardless of the chart, then we either need to have one single profile-openshift.yaml file - which we have today, by copying them - or we'd need the profile files to be independent, not a copy, right?

Sorry, I'm still trying to understand this new helm profile thing...

Reply:
In the profile we can have:

```yaml
global:
  platform: openshift
platform: openshift
```

If it ever becomes enough that we cannot use one profile for all charts (which is plausible, given the chart values are not correlated), we can make machinery to have per-chart profiles. Its just easier _for development_ to keep them the same

Reply:
ok, so we can "duplicate" this info in the openshift profile for now. Do you think long term we should get rid of global as in https://github.com/istio/istio/pull/50113 ?

Reply:
done